src/queue.c

```diff
@@ -5762,7 +5762,25 @@ DISPATCH_NOINLINE DISPATCH_NORETURN
 static void
 _dispatch_sigsuspend(void)
 {
-	static const sigset_t mask;
+	sigset_t mask;
+	int r;
+
+	r = sigemptyset(&mask);
+	(void)dispatch_assume_zero(r);
+
+#if DISPATCH_EVENT_BACKEND_EPOLL
+	// If there are signals that the user may have reasonably
```

Comment:
I think the better solution is to handle this lazily in the code that creates the unotes for signals: when a dispatch source of type SIGNAL is created for a given signal, making a `sigprocmask(SIG_BLOCK, ...)` and document that it does this in the headerdoc for DISPATCH_SOURCE_TYPE_SIGNAL on linux.

Here you're just fishing and there are other signals that are useful that you don't cover which is the part I don't like.